climpred/relative_entropy.py

```diff
@@ -0,0 +1,291 @@
+        ntime = initialized.time.size
+
+    non_spatial_dims = []
+    for dim in control_uninitialized.dims:
+        if dim not in ['x','y','lon','lat','area','period']:
+            non_spatial_dims.append(dim)
+    if not anomaly_data:  # if ds, control are raw values
+        if detrend_by_control_unitialized:
+            anom_x = initialized - control_uninitialized.mean(non_spatial_dims)
+            anom_b = control_uninitialized - control_uninitialized.mean(non_spatial_dims)
+    else:  # leave as is when already anomalies
+        anom_x = initialized
+        anom_b = control_uninitialized
+
+    initializations = initialized.initialization.values
+    length = initialized.time.size
+    iterables = [['R', 'S', 'D'], initializations]
+    mindex = pd.MultiIndex.from_product(
+        iterables, names=['component', 'initialization'])
+    tt = np.arange(1, length)
+    rel_ent = pd.DataFrame(index=tt, columns=mindex)
+    rel_ent.index.name = 'Lead Year'
+    if curv:  # if curvilinear lon(x,y), lat(x,y) data inputs
+        wgts = None
+    else:
+        coslat = np.cos(np.deg2rad(anom_x.coords['lat'].values))
+        wgts = np.sqrt(coslat)[..., np.newaxis]
+    # stack all dimensions member and initialization into time, make time first
+    transpose_dims = list(control_uninitialized.dims)
+    transpose_dims.remove('member')
+    transpose_dims.remove('initialization')
+    dims=tuple(transpose_dims)
+    base_to_calc_eofs = control_uninitialized.stack(
+            new=tuple(non_spatial_dims)).rename({'new':'time'}).set_index(
+            {'time':'time'}).transpose(*dims)
+
+    solver = Eof(base_to_calc_eofs, weights=wgts)
+
+    for init in initializations:
+        for t in initialized.time.values[:ntime]:
+            # P_b base distribution
+            pc_b = solver.projectField(anom_b.sel(initialization=init, time=t)
+                                             .drop('time')
+                                             .rename({'member': 'time'}),
+                                       neofs=neofs, eofscaling=0,
+                                       weighted=False)
+
+            mu_b = pc_b.mean('time')
+            sigma_b = xr.DataArray(np.cov(pc_b.T))
+
+            # P_x initialization distribution
+            pc_x = solver.projectField(anom_x.sel(initialization=init, time=t)
+                                             .drop('time')
+                                             .rename({'member': 'time'}),
+                                       neofs=neofs, eofscaling=0,
+                                       weighted=False)
+
+            mu_x = pc_x.mean('time')
+            sigma_x = xr.DataArray(np.cov(pc_x.T))
+
+            r, d, s = _relative_entropy_formula(sigma_b, sigma_x, mu_x, mu_b,
+                                                neofs)
+
+            rel_ent.T.loc['R', init][t] = r
+            rel_ent.T.loc['D', init][t] = d
+            rel_ent.T.loc['S', init][t] = s
+    return rel_ent
```

Comment:
Can you convert this back to an xarray dataset? You can do this probably through `.from_dataframe()`
(http://xarray.pydata.org/en/stable/pandas.html). Since we are pushing the use of xarray for this package, it would be weird to have some functions return pandas objects.